x-pack/plugins/alerting/server/authorization/alerting_authorization.ts

```diff
@@ -138,6 +142,41 @@ export class AlertingAuthorization {
     return this.authorization?.mode?.useRbacForRequest(this.request) ?? false;
   }
 
+  public async getAuthorizedAlertsIndices(featureIds: string[]): Promise<string[] | undefined> {
```

Comment:
What do you think of moving the bulk of this functionality to the RAC alerts client? I don't think the mapping between feature id and alerts as data index necessarily belongs in the alerting auth client.

What about exposing a function that just wraps `augmentRuleTypesWithAuthorization` like

```
public async getAuthorizedRuleTypesForFeatureIds(featureIds: Set<string>, operations: Array<ReadOperations | WriteOperations>, authorizationEntity: AlertingAuthorizationEntity) {
  return await this.augmentRuleTypesWithAuthorization(
    this.alertTypeRegistry.list(),
    operations,
    authorizationEntity,
    featureIds
  );
}
```

And then move the rest of this function into the RAC alerts client.

Reply:
Great point. Moved this over to our client.